src/contracts/Protocol/BountyQuestion.sol

```diff
@@ -9,63 +9,40 @@ import "@openzeppelin/contracts/access/Ownable.sol";
-//   Voting threshold -> do we want manual approval? -> yes
-//   Admin approval (Messari group wants to write their own challenges)
-// TODO ability for author to unlock and "kill" their question -> reimburse voters
-
-// philosphy -> start out gated and the protocol can evolve in the same way the dao does
-
 /// @custom:security-contact contracts@metricsdao.xyz
-contract BountyQuestion is ERC721, ERC721Enumerable, ERC721URIStorage, ERC721Burnable, Ownable, OnlyApi {
+contract BountyQuestion is Ownable, OnlyApi {
     using Counters for Counters.Counter;
 
     Counters.Counter private _tokenIdCounter;
-    mapping(address => QuestionData[]) public authors; // TODO if we want questions to be transferable, then owner != author
-    mapping(uint256 => uint256) private _createdAt;
 
-    constructor() ERC721("MetricsDAO Question", "MDQ") {
+    // This maps the author to the list of token IDs they have created
+    mapping(address => uint256[]) public authors;
+
+    // This maps the token ID to the question data
+    mapping(uint256 => QuestionData) public questions;
+
+    constructor() {
         _tokenIdCounter.increment();
     }
 
-    // working standard metadata format:  Title, Description, Program
-    function safeMint(address to, string calldata uri) public onlyApi returns (uint256) {
+    function createQuestion(address author, string calldata uri) public onlyApi returns (uint256) {
         uint256 tokenId = _tokenIdCounter.current();
         _tokenIdCounter.increment();
-        _safeMint(to, tokenId);
-        _setTokenURI(tokenId, uri);
-        QuestionData memory _question = QuestionData({tokenId: tokenId, url: uri});
-        authors[to].push(_question);
-        return tokenId;
-    }
 
-    //------------------------------------------------------ Solidity Overrides
-
-    function _beforeTokenTransfer(
-        address from,
-        address to,
-        uint256 tokenId
-    ) internal override(ERC721, ERC721Enumerable) {
-        super._beforeTokenTransfer(from, to, tokenId);
-    }
-
-    function _burn(uint256 tokenId) internal override(ERC721, ERC721URIStorage) {
-        super._burn(tokenId);
-    }
-
-    function tokenURI(uint256 tokenId) public view override(ERC721, ERC721URIStorage) returns (string memory) {
-        return super.tokenURI(tokenId);
-    }
-
-    function supportsInterface(bytes4 interfaceId) public view override(ERC721, ERC721Enumerable) returns (bool) {
-        return super.supportsInterface(interfaceId);
+        questions[tokenId] = QuestionData({tokenId: tokenId, url: uri});
+        // QuestionData memory _question = QuestionData({tokenId: tokenId, url: uri});
+        authors[author].push(tokenId);
+        return tokenId;
     }
 
     function getAuthor(address user) public view returns (QuestionData[] memory) {
-        return authors[user];
+        uint256[] memory created = authors[user];
+
+        QuestionData[] memory ret = new QuestionData[](created.length);
```

Comment:
it's a view function though!  I don't know how that works.  I'll do some reading.  I like the `QuestionData` struct you introduced, and I want that to be the primary data structure because this is a question.. ya know?  Like in a sql database we would have a table of these with a foreign key relationship to another table mapping authors.